spock-core/src/main/java/org/spockframework/mock/runtime/MockMakerRegistryConfiguration.java

```diff
@@ -0,0 +1,29 @@
+package org.spockframework.mock.runtime;
+
+import org.spockframework.util.Beta;
+import org.spockframework.util.Nullable;
+import spock.config.ConfigurationObject;
+
+/**
+ * Configuration settings for the {@link IMockMaker} instances.
+ *
+ * <p>Example:
+ * <pre><code>
+ * mockMaker {
+ *   preferredMockMaker = "your-preferred-mock-maker"
+ * }
+ * </code>
+ * </pre>
+ *
+ * @since 2.4
+ */
+@Beta
+@ConfigurationObject("mockMaker")
+public class MockMakerRegistryConfiguration {
+
+  /**
+   * The preferredMockMaker {@link IMockMaker} instance to use, when creating mock objects.
+   */
+  @Nullable
+  public String preferredMockMaker = null;
```

Comment:
Done.

Reply:
Sorry for intervening here, but are you sure that is a good idea @leonard84?
Now you have to either import the class in the config script or use the FQCN, that's probably ok.

But your intention

> That way one could have some custom default settings for mock makers that support them.

is not fulfilled, is it?

WIth that change now done, you can indeed configure a mock maker if it has additional settings for when it is used as preferred mock maker. But you can also only do this on the preferred mock maker and only if you make one the preferred. Was that really the intention? Then probably just re-close this conversation.

Or was the thought that you can configure settings for all available mock makers that have settings, whether they are used at their default priority or as preferred mock maker? In that case it might be better to have some other way to set these settings and change the preferred mock-maker setting back to a String property maybe?

Reply:
It was what I intended.

> Now you have to either import the class in the config script or use the FQCN, that's probably ok.

Yeah, I think that is fine, it also makes it a bit more obvious where it is coming from.

> With that change now done, you can indeed configure a mock maker if it has additional settings for when it is used as preferred mock maker. But you can also only do this on the preferred mock maker and only if you make one the preferred. Was that really the intention? 

My thinking was this, when setting the default mock maker you don't have an opportunity to configure it in the actual test. For every other mock maker, you have to pass the config anyway in the test, you can just reference some shared custom config there. If you had a way to set additional config for non-default mock makers, then we'd have the problem of merging the default and use-site config, or delegate that to the mock maker.

Reply:
> For every other mock maker, you have to pass the config anyway in the test

Not necessarily.
The mock makers are tried one after the other until one is found that can create the mock if none is given explicitly.
The default one is just the first one tried.
So it could indeed make sense to be able to define default settings for mock makers that are not the default mock maker.

But this can of course also be done in a separate PR if it would make sense.
Just wanted to double-check it is into what you intended. :-)